codec-http/src/main/java/io/netty/handler/codec/http/DefaultFullHttpResponse.java

```diff
@@ -33,31 +33,32 @@ public DefaultFullHttpResponse(HttpVersion version, HttpResponseStatus status) {
     }
 
     public DefaultFullHttpResponse(HttpVersion version, HttpResponseStatus status, ByteBuf content) {
-        this(version, status, content, false);
+        this(version, status, content, true);
     }
 
     public DefaultFullHttpResponse(HttpVersion version, HttpResponseStatus status, boolean validateHeaders) {
-        this(version, status, Unpooled.buffer(0), validateHeaders, false);
+        this(version, status, Unpooled.buffer(0), validateHeaders, true);
     }
 
     public DefaultFullHttpResponse(HttpVersion version, HttpResponseStatus status, boolean validateHeaders,
-                                   boolean singleFieldHeaders) {
-        this(version, status, Unpooled.buffer(0), validateHeaders, singleFieldHeaders);
+                                   boolean allowDuplicateNameHeaders) {
+        this(version, status, Unpooled.buffer(0), validateHeaders, allowDuplicateNameHeaders);
     }
 
     public DefaultFullHttpResponse(HttpVersion version, HttpResponseStatus status,
-                                   ByteBuf content, boolean singleFieldHeaders) {
-        this(version, status, content, true, singleFieldHeaders);
+                                   ByteBuf content, boolean validateHeaders) {
+        this(version, status, content, validateHeaders, true);
     }
 
     public DefaultFullHttpResponse(HttpVersion version, HttpResponseStatus status,
-                                   ByteBuf content, boolean validateHeaders, boolean singleFieldHeaders) {
-        super(version, status, validateHeaders, singleFieldHeaders);
+                                   ByteBuf content, boolean validateHeaders, boolean allowDuplicateNameHeaders) {
+        super(version, status, validateHeaders, allowDuplicateNameHeaders);
         if (content == null) {
             throw new NullPointerException("content");
```

Comment:
I will use Netty's checkNotNull (we still support 1.6 and 1.7 last I checked :) )